### PR TITLE
Add minimal Ascend NPU device parsing to select_device

### PR DESCRIPTION
**Pull request by GiantAxeWhy**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Adds minimal Huawei Ascend NPU device parsing to `select_device`, enabling basic `npu` device selection in Ultralytics. 🚀

### 📊 Key Changes
- Adds Ascend NPU handling in `ultralytics/utils/torch_utils.py` within `select_device`.
- Normalizes device strings by removing spaces earlier with `replace(" ", "")` before existing cleanup.
- Detects `npu` device requests such as `npu`, `npu:0`, and `npu:0,1`.
- Imports `torch_npu` and validates that `torch.npu` is available before proceeding.
- Sets the active NPU device with `torch.npu.set_device(idx)` and returns `torch.device(f"npu:{idx}")`.
- Logs the selected NPU index and device name when verbose mode is enabled.
- Raises a clear error message when an invalid or unavailable Ascend NPU is requested.

### 🎯 Purpose & Impact
- Enables basic Ascend NPU support for users running Ultralytics on Huawei AI hardware. ⚡
- Improves device parsing flexibility with minimal, targeted changes to existing selection logic.
- Makes NPU usage more intuitive by supporting common input formats without requiring manual setup.
- Helps broaden hardware compatibility, though this PR appears focused on initial parsing and selection rather than full multi-device NPU workflows.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @GiantAxeWhy, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review, and an engineer will assist you soon. Please review the checklist below to help ensure a smooth integration of your Ascend NPU device parsing update. 😊

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/utils/torch_utils.py [hunk 1]
```diff
@@ -164,10 +164,33 @@ def select_device(device="", newline=False, verbose=True):
...
+    if device.startswith("npu"):
+        import torch_npu  # noqa
+
+        if not hasattr(torch, "npu") or not torch.npu.is_available():
+            raise ValueError(
+                f"Invalid NPU 'device={device}' requested. "
+                "Use 'device=cpu' or provide a valid Ascend NPU device, e.g. 'npu:0'."
+            )
+
+        # support: npu / npu:0 / npu:0,1
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/torch_utils.py]: ⚠️ **HIGH**: The comment says `npu:0,1` is supported, but this branch silently discards every index after the first and returns a single `torch.device`. That can mislead callers into thinking multi-NPU selection succeeded when only `npu:0` was activated. Either reject comma-separated NPU requests explicitly or add real multi-device handling.

File: ultralytics/utils/torch_utils.py [hunk 2]
```diff
@@ -164,10 +164,33 @@ def select_device(device="", newline=False, verbose=True):
...
     s = f"Ultralytics {__version__} 🚀 Python-{PYTHON_VERSION} torch-{TORCH_VERSION} "
-    device = str(device).lower()
-    for remove in "cuda:", "none", "(", ")", "[", "]", "'", " ":
+    device = str(device).lower().replace(" ", "")
+    for remove in "cuda:", "none", "(", ")", "[", "]", "'":
         device = device.replace(remove, "")  # to string, 'cuda:0' -> '0' and '(0, 1)' -> '0,1'
 
+    # Huawei Ascend NPU
+    if device.startswith("npu"):
+        import torch_npu  # noqa
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/torch_utils.py]: 💡 **MEDIUM**: `import torch_npu` will raise a raw `ImportError` when the package is not installed, so `device='npu'` fails before your more helpful validation/error path runs. Wrapping this import and re-raising a `ValueError` here would keep NPU selection failures consistent with the rest of `select_device`.

File: ultralytics/utils/torch_utils.py [hunk 3]
```diff
@@ -164,10 +164,33 @@ def select_device(device="", newline=False, verbose=True):
...
+
+        if not hasattr(torch, "npu") or not torch.npu.is_available():
+            raise ValueError(
+                f"Invalid NPU 'device={device}' requested. "
+                "Use 'device=cpu' or provide a valid Ascend NPU device, e.g. 'npu:0'."
+            )
+
+        # support: npu / npu:0 / npu:0,1
+        if ":" in device:
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/torch_utils.py:195]: 💡 **MEDIUM**: This accepts any string starting with `npu` and then falls through to `int(first)`. Inputs like `npu:`, `npu:x`, or even `npufoo` will raise unclear exceptions or be misparsed as device 0. Adding format validation here would make invalid-device errors deterministic and much easier to diagnose.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The NPU parsing is a useful minimal addition, but there are a few correctness gaps to address before merging: missing-package handling currently raises a raw import error, comma-separated NPU IDs are advertised as supported but only the first device is used, and malformed `npu` strings are not validated cleanly.

💬 Posted 3 inline comments

**Comment by GiantAxeWhy** [comment]
I have read the CLA Document and I sign the CLA
